src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1096,6 +1148,25 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         _terminal->SetCursorVisible(!_terminal->IsCursorVisible());
     }
 
+    // Method Description:
+    // - Sets selection's end position to match supplied cursor position, e.g. while mouse dragging.
+    // Arguments:
+    // - cursorPosition: in pixels
+    void TermControl::_SetEndSelectionPointAtCurstor(const Windows::Foundation::Point cursorPosition)
+    {
+        auto terminalPosition = _GetTerminalPosition(cursorPosition);
+
+        const short lastVisibleRow = std::max(_terminal->GetViewport().Height() - 1, 0);
+        const short lastVisibleCol = std::max(_terminal->GetViewport().Width() - 1, 0);
+
+        terminalPosition.Y = std::clamp(terminalPosition.Y, short{ 0 }, lastVisibleRow);
+        terminalPosition.X = std::clamp(terminalPosition.X, short{ 0 }, lastVisibleCol);

```

Comment:
Check out `Viewport::Clamp()`. Is there a way we could use that here?

Reply:
`Viewport::Clamp()` expects absolute position, whereas src and dst of `terminalPosition` are relative to scroll offset.